lib/Doctrine/Common/DataFixtures/Executor/ORMExecutor.php

```diff
@@ -25,12 +25,14 @@ class ORMExecutor extends AbstractExecutor
     /** @var ORMReferenceListener */
     private $listener;
 
+    /** @var bool */
+    private $singleTransaction;
+
     /**
-     * Construct new fixtures loader instance.
-     *
-     * @param EntityManagerInterface $em EntityManagerInterface instance used for persistence.
+     * @param EntityManagerInterface $em                EntityManagerInterface instance used for persistence.
+     * @param bool                   $singleTransaction Whether to use a single transaction when loading fixtures.
      */
-    public function __construct(EntityManagerInterface $em, ?ORMPurgerInterface $purger = null)
+    public function __construct(EntityManagerInterface $em, ?ORMPurgerInterface $purger = null, bool $singleTransaction = true)
```

Comment:
Let's avoid a boolean here please. We've seen that there could be at least one more strategy (no transaction at all). Let's maybe instead inject an interface, and write 2 implementations of that interface?

Reply:
> Let's maybe instead inject an interface, and write 2 implementations of that interface?

Before we do that, let's rather do a second `ORMExecutor`. That interface you're suggesting would basically contain the main method of `AbstractExecutor`, `execute()`.

Reply:
Even better 👍

Reply:
Something like `MultipleTransactionsORMExecutor`? There would be quite a lot in common with the `ORMExecutor` and AFAIK we don't want to use inheritance.
Why not having an `ExecutorInterface` as a strategy in the ORMExecutor?

Reply:
Hard to figure out what you mean exactly, can you push something to this PR?

Reply:
I pushed a PoC, it's a bit in the idea of having an enum-like while keeping the logic in classes.

Reply:
This PoC is insanely complex for such a simple change. Please, let's make this a separate executor. Use a trait if you need to share logic between both executors.